File: Code.java
// Definition for singly-linked list:
// class ListNode<T> {
//   ListNode(T x) {
//     value = x;
//   }
//   T value;
//   ListNode<T> next;
// }
//
ListNode<Integer> rearrangeLastN(ListNode<Integer> l, int n) {
    ListNode<Integer> slow = l;
    ListNode<Integer> fast = l;
    ListNode<Integer> res = new ListNode<Integer>(0);
    res.next = l;
    ListNode<Integer> head = res;
    
    int counter = 0;
    
    while(fast != null)
    {
        counter ++;
        
        if(fast.next != null)
        {
            fast = fast.next;
        }
        else{
            break;
        }
    }
    
    System.out.println(counter);

    if(n == 0 || counter <= n || counter == 0)
        return l;
    
    fast.next = l;
    
    for(int i = 0; i < counter - n; i++)
    {
        fast = fast.next;
        res.next = res.next.next;   
    }
    
    
    fast.next = null;
    
    return res.next;
}

File: Code.c
//
// Definition for singly-linked list:
// typedef struct list_##name {
//   type value;
//   struct list_##name *next;
// } list_##name;
//
// list_##name *alloc_list_##name(type v) {
//   list_##name *l = calloc(1, sizeof(list_##name));
//   l->value = v;
//   return l;
// }
//
//

void print_linked_list(list_integer * list){
    while(list){
        printf("%d ", list->value);
        list = list->next;
    }
}

list_integer * rearrangeLastN(list_integer * l, int n) {  
    list_integer * tmp = l;
    int count;
    
    if(l == NULL)
        return NULL;
    
    while(tmp != NULL){
        count++;
        if(tmp->next != NULL)
            tmp = tmp->next;
        else
            break;
    }
    
    if(n == 0 && count <= n && count == 0)
        return l;
    
    tmp->next = l;
    
    for(int i = 0; i < count - n; i++)
        tmp = tmp->next;
    
    l = tmp->next;
    tmp->next = NULL;
    
    return l;
}
